yeat/tests/test_cli.py

```diff
@@ -120,3 +121,55 @@ def test_custom_downsample_input(
     assert df.iloc[12]["sample_contigs"] == num_contigs
     assert df.iloc[13]["sample_contigs"] == largest_contig
     assert df.iloc[14]["sample_contigs"] == total_len
+
+
+@pytest.mark.parametrize("coverage", [("-1"), ("0")])
+def test_invalid_custom_coverage_negative(coverage):
+    arglist = [
+        data_file("megahit.cfg"),
+        data_file("short_reads_1.fastq.gz"),
+        data_file("short_reads_2.fastq.gz"),
+        "--coverage",
+        coverage,
+    ]
+    with pytest.raises(SystemExit) as error:
+        args = cli.get_parser().parse_args(arglist)
+    assert isinstance(error.value.__context__, argparse.ArgumentError)
+    assert f"{coverage} is not a positive integer" in error.value.__context__.message
+
+
+@pytest.mark.parametrize("coverage", [("string"), ("3.14")])
+def test_invalid_custom_coverage_noninteger(coverage):
+    arglist = [
+        data_file("megahit.cfg"),
+        data_file("short_reads_1.fastq.gz"),
+        data_file("short_reads_2.fastq.gz"),
+        "--coverage",
+        coverage,
+    ]
+    with pytest.raises(SystemExit) as error:
+        args = cli.get_parser().parse_args(arglist)
+    assert isinstance(error.value.__context__, argparse.ArgumentError)
+    assert f"{coverage} is not an integer" in error.value.__context__.message
```

Comment:
Like the test above, since I am now raising a `ArgumentError` for "not int" values, I'm doing the same here.

It seems like there are a few comments on the recent question you posted on `pytest`'s GitHub Q&A.

https://github.com/pytest-dev/pytest/discussions/10440

Reply:
From what I'm seeing, the `exit_on_error` flag looks the best.

Reply:
UPDATE: Added the new flag and the tests were consolidated down to a regex match.